src/formats/gif.zig

```diff
@@ -287,7 +287,7 @@ pub const GIF = struct {
             return Image.ReadError.InvalidData;
         }
 
-        const global_color_table_size = @as(usize, 1) << (@as(u6, @intCast(self.header.flags.global_color_table_size)) + 1);
+        const global_color_table_size = @as(usize, 1) << (@as(if (@sizeOf(usize) == 4) u5 else u6, @intCast(self.header.flags.global_color_table_size)) + 1);
```

Comment:
this could be put inside a constant
```zig
const ColorTableShiftType = if (@sizeOf(usize) == 4) u5 else u6;
```

So you don't have to do the `if` logic everytime you bit shift